libqtile/hook.py

```diff
@@ -850,6 +850,53 @@ def lock_on_sleep():
         inhibitor.want_sleep()
         return self._subscribe("suspend", func)
 
+    def custom(self, hook_name):
+        """
+        Use to create user-defined hooks.
+
+        The purpose of these hooks is to allow a hook to be fired by an external application.
+
+        Hooked functions can receive arguments but it is up to the application firing the hook to ensure
+        the correct arguments are passed. No checking will be performed by qtile.
+
+        Example:
+
+        .. code:: python
+
+          from libqtile import hook
+          from libqtile.log_utils import logger
+
+          @hook.subscribe.custom("my_custom_hook")
+          def hooked_function():
+            logger.warning("Custom hook received.")
+
```

Comment:
that's a super nice api

Reply:
I actually wonder whether it would be neater to do:
```python
@hook.subscribe.custom.my_custom_hook
def hooked_function():
   ...
```

Reply:
Would `user` be better than `custom`?

Reply:
Both work really. Do we really need a prefix anyway?

Reply:
No. You don't need the prefix. I just thought it made things a bit clearer and prevented any clashes with existing hooks.

Reply:
If it's all the same to you guys, I do think using a string is clearer here. Then it's clear that it's an arbitrary piece of text (and supports punctation, spaces etc) rather than an attribute, which could be interpreted as being one of a predefined set of names. Bit less magical

Reply:
Killing the magic :(

I'm fine to change it back!